packages/devtools_app/lib/src/service_manager.dart

```diff
@@ -275,6 +276,8 @@ class ServiceConnectionManager {
     vm = null;
     sdkVersion = null;
     connectedApp = null;
+    generateDevToolsTitle();
```

Comment:
I would either move the devtools title functionality directly into service_manager or add logic to title so it is just listening for events on the service manager.

Reply:
That would make it challenging to modify titles inside google3. 

We also need to know what the connected app type is, which is initialized in the ConnectedApp class. This class doesn't currently dispatch any events. We could change it so that it does, but I'm not sure what extra value that would add. We only initialize connected app values on vm service opened, and we clear out the connected app here in vm service closed. I suppose we could add a method to ConnectedApp `detach` or similar, that calls `generateDevToolsTitle` but that feels like a lateral change without much effect.